examples/datagen_with_al/README.md

```diff
@@ -0,0 +1,221 @@
+# Training with Active Learning
+
+## Installation
+
+The following steps require PrimeQA to be [installed](../../README.md#Installation).
+
+## Prerequisites
+
+Before we start, I suggest to set the cache directories unless you have enough space in `$HOME/.cache`:
+```bash
+CACHE_DIR = path/to/dir # set this to a custom directory where you want to have huggingface downloads cached
+export TRANSFORMERS_CACHE=$CACHE_DIR
+export HF_DATASETS_CACHE=$CACHE_DIR
+export HF_MODULES_CACHE=$CACHE_DIR
+```
+
+If you use CUDA, make sure to also set `CUDA_VISIBLE_DEVICES` to the appropriate device.
+E.g. for GPU 2 (the third one) on the current host:
+```bash
+export CUDA_VISIBLE_DEVICES=2
```

Comment:
Do we need to make sure that it always runs on gpu 2? If not let's change it to export CUDA_VISIBLE_DEVICES=<gpu_device_id> . can we automatically set it in the python code itself based on device availability instead of asking user to set env variable?